File: java/year_2024/Day2/Day2.java
package year_2024.Day2;

import java.util.ArrayList;

import static util.FileUtils.readFile;

public class Day2 {

    private static int Part1(ArrayList<String> lines) {
        for (String line : lines) {
            // Parse strings to ints
            String[] levels = line.split("\\s+");
            int[] nums = new int[levels.length];
            for (int i = 0; i < levels.length; i++) {
                nums[i] = Integer.parseInt(levels[i]);
            }

            boolean decreasing;
            if (nums[0] > nums[1]) {
                decreasing = true;
            } else if (nums[0] < nums[1]) {
                decreasing = false;
            }

            for (int i = 0; i < nums.length; i++) {

            }
        }
        return 0;
    }

    public static void main(String[] args) {
        ArrayList<String> lines = readFile("C:\\Users\\josep\\Desktop\\Advent of Code\\2024\\Day2\\Example.txt");
        System.out.println(Part1(lines));
    }
}

File: java/year_2022/Day2/Day2.py
round = [0, 0]
score = 0
f = open(r"C:\\Users\\josep\Desktop\Advent of Code\\2022\Day2.txt", "r")

def calcWin(round):
    global score
    if(round[0] == "A"):
        if(round[1] == "X"):
            score += 3
        elif(round[1] == "Y"):
            score += 4
        elif(round[1] == "Z"):
            score += 8
    elif(round[0] == "B"):
        if(round[1] == "X"):
            score += 1
        elif(round[1] == "Y"):
            score += 5
        elif(round[1] == "Z"):
            score += 9
    else:
        if(round[1] == "X"):
            score += 2
        elif(round[1] == "Y"):
            score += 6
        elif(round[1] == "Z"):
            score += 7

for lines in f:
    round[0] = lines[0]
    round[1] = lines[2]
    calcWin(round)

print(score)
f.close()